### ISSUE TITLE
Make vocabSize defaults as powers of 2 to speed up DeepLearning 

### ISSUE THREAD

**Issue by rsindreu** [huggingface/tokenizers#1090]
Currently most tokenizers have multiples of 10 as vocabSize (ex: BPE: 30000, UNI: 8000, WPC: 30000).

From this post: https://towardsdatascience.com/optimize-pytorch-performance-for-speed-and-memory-efficiency-2022-84f453916ea6

[These experiments](https://docs.nvidia.com/deeplearning/performance/dl-performance-fully-connected/index.html) have shown setting the output dimension and the batch size as the multiples of 8 (i.e., 33712, 4088, 4096) can accelerate the computation by 1.3x to 4x compared to the output dimension of 33708 and the batch sizes of 4084 and 4095, which are indivisible by 8. The acceleration magnitude depends on the process types (e.g., forward passes or the gradient calculations) and the cuBLAS versions. Especially, if you work on NLP, remember to check your output dimension, which is usually the vocabulary size.

Can we change these defaults to powers of 2? I.e. 32768 or 2^15 instead of 30000, 8192 or 2^13 instead of 8000.

**Comment by rsindreu** [comment]
Turns out these are already multiples of 8, although not powers of 2, the benefit is not as clear.

**Comment by Narsil** [comment]
Hi @rsindreu you're entirely right ! 

Tbh, the speedups and the exact numbers might depend on hardware. Powers of two is the safest bet, but quite limiting. Multiples of 8 is not perfect either.

For future readers I'd like to point out that the "special_tokens" that might get added after training do count in the final vocabulary size.
